dcap-ql/src/quote.rs

```diff
@@ -284,10 +395,19 @@ impl<'a> Quote3SignatureEcdsaP256<'a> {
         &self.signature
     }
 
-    pub fn attestation_public_key(&self) -> &[u8] {
+    pub fn raw_attestation_public_key(&self) -> &[u8] {
         &self.attestation_public_key
     }
 
+    #[cfg(feature = "verify")]
+    fn attestation_public_key(&self) -> Result<Pk> {
+        let mut pt = vec![0x4];
+        pt.append(&mut self.raw_attestation_public_key().to_vec());
```

Comment:
Converting to a Vec and then appending to a Vec is doing one unnecessary memcpy. Use `extend_from_slice`.